cmd/run.go

```diff
@@ -113,14 +115,25 @@ func (c *cmdRun) run(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+	engine.AbortFn = runSubAbort
 
 	// Spin up the REST API server, if not disabled.
 	if c.gs.flags.address != "" {
 		initBar.Modify(pb.WithConstProgress(0, "Init API server"))
+
+		apiWG := &sync.WaitGroup{}
+		apiWG.Add(2)
+		defer apiWG.Wait()
+
+		srvCtx, srvCancel := context.WithCancel(globalCtx)
+		defer srvCancel()
+
+		// TODO: send the ExecutionState and MetricsEngine instead of the Engine
```

Comment:
Reminder to delete this comment, since it's now obsolete

Reply:
Resolved by the [latest force-push](https://github.com/grafana/k6/compare/96162e1ab53e659635b53cef8717d95f4bb871cc..30908a1fe3e691111f91a520cfcb2e6f7f2c7a52)